resources/nspawn.go

```diff
@@ -0,0 +1,351 @@
+
+	// systemd "github.com/coreos/go-systemd/dbus" // change namespace
+	systemdUtil "github.com/coreos/go-systemd/util"
+	"github.com/godbus/dbus" // namespace collides with systemd wrapper
+	machined "github.com/joejulian/go-systemd/machine1"
+)
+
+const (
+	running               = "running"
+	stopped               = "stopped"
+	dbusInterface         = "org.freedesktop.machine1.Manager"
+	dbusPath              = "/org/freedesktop/machine1"
+	machineNew            = "org.freedesktop.machine1.Manager.machineNew"
+	machineRemoved        = "org.freedesktop.machine1.Manager.machineRemoved"
+	nspawnServiceTemplate = "systemd-nspawn@%s"
+)
+
+func init() {
+	gob.Register(&NspawnRes{})
+}
+
+// NspawnRes is an nspawn container resource
+type NspawnRes struct {
+	BaseRes `yaml:",inline"`
+	State   string `yaml:"state"`
+}
+
+// Init runs some startup code for this resource.
+func (obj *NspawnRes) Init() error {
+	obj.BaseRes.kind = "Nspawn"
+	return obj.BaseRes.Init()
+}
+
+// NewNspawnRes is the constructor for this resource.
+func NewNspawnRes(name string, state string) (*NspawnRes, error) {
+	obj := &NspawnRes{
+		BaseRes: BaseRes{
+			Name: name,
+		},
+		State: state,
+	}
+	return obj, obj.Init()
+}
+
+// Validate params
+func (obj *NspawnRes) Validate() error {
+	switch obj.State {
+	case running, stopped:
+		break
+	default:
+		return fmt.Errorf("Invalid State: %s", obj.State)
+	}
+	return nil
+}
+
+// Watch for state changes and sends a message to the bus if there is a change.
+func (obj *NspawnRes) Watch(processChan chan event.Event) error {
+	// Boilerplate
+	// TODO: Boilerplate sucks. Find a way to prevent its necessity.
+	if obj.IsWatching() {
+		return nil
+	}
+	obj.SetWatching(true)
+	defer obj.SetWatching(false)
+	cuid := obj.converger.Register()
+	defer cuid.Unregister()
+
+	var startup bool
+	Startup := func(block bool) <-chan time.Time {
+		if block {
+			return nil // blocks forever
+		}
+		return time.After(time.Duration(500) * time.Millisecond) // 1/2 the resolution of converged timeout
+	}
+	// End Boilerplate
+
+	// This resource depends on systemd. Ensure that it's running.
+	// TODO: If systemd is not running, should we start it?
+	if !systemdUtil.IsRunningSystemd() {
+		return fmt.Errorf("Systemd is not running.")
+	}
+
+	// Create a private message bus
+	bus, err := util.SystemBusPrivateUsable()
+	if err != nil {
+		return fmt.Errorf("Failed to connect to bus: %s", err)
+	}
+
+	// Add a match rule to match messages going through the message bus.
+	bus.BusObject().Call("org.freedesktop.DBus.AddMatch", 0,
+		"type='signal',interface='org.freedesktop.machine1.Manager',eavesdrop='true'")
+	buschan := make(chan *dbus.Signal, 10)
+	bus.Signal(buschan)
+
+	var send = false
+	var exit = false
+	// var dirty = false
+	// var invalid = false
+	// var activeSet = false
+	// var previous bool
+
+	for {
+		obj.SetState(ResStateWatching)
+		select {
+		// If this resource has been idle for long enough, set converged to allow
```

Comment:
If you wish to keep this comment, please wrap it at 80 chars because it fits nicely that way.